tests/api/utils/test_rendering.py

```diff
@@ -63,3 +70,68 @@ def test_that_undefined_variables_raise_error():
 
 def test_contex_can_access_environment_variable():
     assert snowflake_cli_jinja_render("&{ ctx.env.USER }") == os.environ.get("USER")
+
+
+def test_resolve_variables_in_project_no_cross_variable_dependencies():
+    pdf = ProjectDefinition(
+        env={
+            "number": 1,
+            "string": "foo",
+            "boolean": True,
+        }
+    )
+    result = _add_project_context({}, project_definition=pdf)
+    assert result == {
+        "ctx": ProjectDefinition(
+            definition_version="1.1",
+            native_app=None,
+            snowpark=None,
+            streamlit=None,
+            env={"number": 1, "string": "foo", "boolean": True},
+        )
+    }
+
+
+def test_resolve_variables_in_project_cross_variable_dependencies():
+    pdf = ProjectDefinition(
+        env={
+            "A": 42,
+            "B": "b=&{ ctx.env.A }",
+            "C": "&{ ctx.env.B } and &{ ctx.env.A }",
+        }
+    )
+    result = _add_project_context({}, project_definition=pdf)
+    assert result == {
+        "ctx": ProjectDefinition(
+            definition_version="1.1",
+            native_app=None,
+            snowpark=None,
+            streamlit=None,
+            env={"A": 42, "B": "b=42", "C": "b=42 and 42"},
+        )
+    }
```

Comment:
@sfc-gh-cgorrie @sfc-gh-melnacouzi if we can agree to use `key:value` format for `env` schema then this looks pretty promising. Also